testing-suite/staking-v4/get_infos/get_delegation_info.py

```diff
@@ -0,0 +1,25 @@
+import requests
+from config import DEFAULT_PROXY
+from helpers import base64_to_hex
+from multiversx_sdk_core import Address
+
+
+def get_delegation_contract_address_from_tx(tx_hash):
+
+    response = requests.get(f"{DEFAULT_PROXY}/transaction/{tx_hash}?withResults=True")
+    response.raise_for_status()
+    parsed = response.json()
+
+    general_data = parsed.get("data")
+    transaction_data = general_data.get("transaction")
+    logs_data = transaction_data.get("logs")
+    events_data = logs_data.get("events")
+    first_set_of_events = events_data[0]
+    topics = first_set_of_events.get("topics")
+    delegation_contract_address = topics[1]
```

Comment:
No. It will return None, which is fine.

Reply:
It will cause run time errors.
This is also CHAT GPT's opinion :) (I asked him exactly on this topic what needs to be done). Please add a task to solve it

>Therefore, your concern about potential runtime errors or panics due to missing data or incorrect indices is valid. If the dataset is not structured as expected or if checks are not put in place to verify that each required piece of data is available before trying to access it, Python will raise exceptions (like IndexError or KeyError), which will stop the program unless they are caught and handled using try-except blocks.